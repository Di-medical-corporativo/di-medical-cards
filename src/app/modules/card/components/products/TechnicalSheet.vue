<template>
  <div class="card-technicals__technicalSheet">
    <img
        class="card-technicals__technicalSheet__image"
        :src="image"
        @click="$emit('openModal', image)"
    >
  </div>
</template>

<script>
export default {
  props: {
    image: {
      type: String
    }
  },
  emits: ['openModal']
}
</script>
