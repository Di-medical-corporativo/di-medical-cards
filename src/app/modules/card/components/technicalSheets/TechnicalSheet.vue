<template>
  <div class="card-technicals__technicalSheet">
    <v-lazy-image
        class="card-technicals__technicalSheet__image"
        :src="image"
        @click="$emit('openModal', image)"
    />
  </div>
</template>

<script>
import VLazyImage from 'v-lazy-image'
export default {
  props: {
    image: {
      type: String
    }
  },
  emits: ['openModal'],
  components: {
    VLazyImage
  }
}
</script>
