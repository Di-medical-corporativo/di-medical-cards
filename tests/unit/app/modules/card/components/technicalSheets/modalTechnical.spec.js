import { shallowMount } from '@vue/test-utils'
import TechnicalSheet from '../../../../../../../src/app/modules/card/components/technicalSheets/ModalTechnical.vue'

jest.mock('vue-router', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn()
  })),
  useRoute: jest.fn(() => ({
    params: {
      sucursal: 'sur'
    }
  }))
}))

describe('Technical sheet modal component', () => {
  test('should match snapshot', () => {
    const wrapper = shallowMount(TechnicalSheet, {
      props: {
        imageToOpen: 'https://firebasestorage.googleapis.com/v0/b/di-medical-del-sur.appspot.com/o/technicalSheets%2F3f0c89b3-c25e-4176-adc7-48fd69263b0c-technical.jpg?alt=media&token=0721fbd8-6891-4915-a084-e66108940c39'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should display image from props property', () => {
    const imageProps = 'https://firebasestorage.googleapis.com/v0/b/di-medical-del-sur.appspot.com/o/technicalSheets%2F3f0c89b3-c25e-4176-adc7-48fd69263b0c-technical.jpg?alt=media&token=0721fbd8-6891-4915-a084-e66108940c39'
    const wrapper = shallowMount(TechnicalSheet, {
      props: {
        imageToOpen: imageProps
      }
    })
    const image = wrapper.find('.card-technicals__modal__background__image')
    expect(image.attributes('src')).toBe(imageProps)
  })

  test('should emit closeModal event when the user clicks the close button', () => {
    const imageProps = 'https://firebasestorage.googleapis.com/v0/b/di-medical-del-sur.appspot.com/o/technicalSheets%2F3f0c89b3-c25e-4176-adc7-48fd69263b0c-technical.jpg?alt=media&token=0721fbd8-6891-4915-a084-e66108940c39'
    const wrapper = shallowMount(TechnicalSheet, {
      props: {
        imageToOpen: imageProps
      }
    })
    const buttonToClose = wrapper.find('.card-technicals__modal__background__close')
    buttonToClose.trigger('click')
    expect(wrapper.emitted('closeModal')).toBeDefined()
  })

  test('should emit closeModal event when the user clicks surrounding area of image', () => {
    const imageProps = 'https://firebasestorage.googleapis.com/v0/b/di-medical-del-sur.appspot.com/o/technicalSheets%2F3f0c89b3-c25e-4176-adc7-48fd69263b0c-technical.jpg?alt=media&token=0721fbd8-6891-4915-a084-e66108940c39'
    const wrapper = shallowMount(TechnicalSheet, {
      props: {
        imageToOpen: imageProps
      }
    })
    const buttonToClose = wrapper.find('.card-technicals__modal__background')
    buttonToClose.trigger('click')
    expect(wrapper.emitted('closeModal')).toBeDefined()
  })
})
