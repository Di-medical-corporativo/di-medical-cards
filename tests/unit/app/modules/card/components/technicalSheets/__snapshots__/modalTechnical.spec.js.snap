// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technical sheet modal component should match snapshot 1`] = `
<div class="card-technicals__modal">
  <div class="card-technicals__modal__background outer">
    <v-lazy-image-stub src="https://firebasestorage.googleapis.com/v0/b/di-medical-del-sur.appspot.com/o/technicalSheets%2F3f0c89b3-c25e-4176-adc7-48fd69263b0c-technical.jpg?alt=media&amp;token=0721fbd8-6891-4915-a084-e66108940c39" srcplaceholder="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" intersectionoptions="[object Object]" usepicture="false" class="card-technicals__modal__background__image inner"></v-lazy-image-stub>
    <div style="background: rgb(33, 141, 155);" class="card-technicals__modal__background__close"><i class="bi bi-x-lg icon"></i></div>
  </div>
</div>
`;
